Dronecode-SDK-Swift/Telemetry.swift

```diff
@@ -381,289 +381,649 @@ public class Telemetry {
     private func createPositionObservable() -> Observable<Position> {
         return Observable.create { observer in
             let positionRequest = DronecodeSdk_Rpc_Telemetry_SubscribePositionRequest()
-
-            do {
-                let call = try self.service.subscribePosition(positionRequest, completion: nil)
-                while let response = try call.receive() {
-                    let position = Position(latitudeDeg: response.position.latitudeDeg, longitudeDeg: response.position.longitudeDeg, absoluteAltitudeM: response.position.absoluteAltitudeM, relativeAltitudeM: response.position.relativeAltitudeM)
-
-                    observer.onNext(position)
+            var call: DronecodeSdk_Rpc_Telemetry_TelemetryServiceSubscribePositionCall?
+            
+            func subscribePosition() -> DronecodeSdk_Rpc_Telemetry_TelemetryServiceSubscribePositionCall? {
+                do {
+                    return try self.service.subscribePosition(positionRequest, completion: { (callResult) in
+                        if callResult.statusCode != .cancelled && callResult.statusCode != .ok {
+                            call = subscribePosition()
```

Comment:
Do I understand it correctly that this is recursion and will grow the stack every time we have to re-subscribe?

Reply:
It looks a little confusing but it's not a classic recursion, it's an asynchronous recursion as we assign a new function inside of the completion handler. It won't increase the stack of calls.